academic_observatory_workflows/workflows/pubmed_telescope.py

```diff
@@ -0,0 +1,1345 @@
+
+        :param entity_type: Type of the record, either "additions" or "deletions".
+        :return: Retuns the path to the transformed file.
+        """
+
+        assert self.changefile_release is not None, "Changefile.transform_file_path: self.changefile_release is None"
+        return os.path.join(self.changefile_release.transform_folder, f"{entity_type}_*.{self.file_type}")
+
+    def merged_transform_file_path(
+        self, entity_type: str, first_file_index: int, last_file_index: int, part_num: int = None
+    ) -> str:
+        """
+        Give path to the merged transform file.
+
+        :param entity_type: Type of the record, either "additions" or "deletions".
+        :return: Retuns the path to the transformed file.
+        """
+
+        assert self.changefile_release is not None, "Changefile.transform_file_path: self.changefile_release is None"
+        if part_num:
+            return os.path.join(
+                self.changefile_release.transform_folder,
+                f"merged_{entity_type}_{first_file_index}-{last_file_index}_part_{part_num}.{self.file_type}",
+            )
+        else:
+            return os.path.join(
+                self.changefile_release.transform_folder,
+                f"merged_{entity_type}_{first_file_index}-{last_file_index}.{self.file_type}",
+            )
+
+    def merged_tranform_file_pattern(self, entity_type: str) -> str:
+        """
+        Return a glob pattern for the merged transform files.
+
+        :param entity_type: Type of the record, either "additions" or "deletions".
+        :return: Retuns the path to the transformed file.
+        """
+        assert self.changefile_release is not None, "Changefile.transform_file_path: self.changefile_release is None"
+        return os.path.join(
+            self.changefile_release.transform_folder,
+            f"merged_{entity_type}_*-*_part_*.{self.file_type}",
+        )
+
+
+class PubmedEntity:
+    def __init__(
+        self,
+        name: str,
+        type: str,
+        sub_key: str,
+        set_key: str,
+        PMID_location: str,
+    ):
+        """
+        Holds the metadata about the records stored in Pubmed.
+
+        :param name: Name of the enitty, articles, book articles or book documents.
+        :param type: Either a set of 'addition' or 'deletion' records.
+        :param sub_key: Sub key location of Pubmed data.
+        :param set_key: Key location of Pubmed.
+        :param PMID_location: Location of PMID values
+        """
+
+        self.name = name
+        self.type = type
+        self.sub_key = sub_key
+        self.set_key = set_key
+        self.PMID_location = PMID_location
+        self.schema_file_path = os.path.join(default_schema_folder(), "pubmed", f"{name}_2023-01-01.json")
+
+
+class PubMedRelease(ChangefileRelease):
+    def __init__(
+        self,
+        *,
+        dag_id: str,
+        run_id: str,
+        cloud_workspace: CloudWorkspace,
+        bq_dataset_id: str,
```

Comment:
I don't think the bq_dataset_id is used by the release anywhere. It shouldn't really be tied to the release anyhow so you can probably remove this